### ISSUE TITLE
publish method not calling callback when consumed RAM is more than 40%

### ISSUE THREAD

**Issue by vermaslal** [amqp-node/amqplib#321]
I am doing a test on my server with 1GB RAM. RabbitMQ using more than 500MB RAM on my server and it not accepting messages as described here http://www.rabbitmq.com/memory.html#memsup

But I am not getting any signal like callback when using publish method like below.

```
console.log('Publishing message');
    try {
        pubChannel.publish(exchange, routingKey, content, {persistent: true},
                function (err, ok) {
                    if (err) {
                        console.error("[AMQP] publish", err);
                        offlinePubQueue.push([exchange, routingKey, content]);
                        pubChannel.connection.close();
                    } else {
                        console.log('Published to queue');
                    }
                });
    } catch (e) {
        console.error("[AMQP] publish", e.message);
        offlinePubQueue.push([exchange, routingKey, content]);
    }
```

How to confirm that my message is published or deferred?

EDIT: However it's accepting message from rabbitmq_management web view panel.

**Comment by kkamkou** [comment]
What is expected? You can observe `bufferSize` or listen `drain` event. see: `blocked`, `unblocked`

**Comment by vermaslal** [comment]
Hi @kkamkou, Thanks for your replay. I will check and update.
